src/katgpucbf/fgpu/engine.py

```diff
@@ -818,8 +819,8 @@ async def run_transmit(self, streams: list["spead2.send.asyncio.AsyncStream"]) -
                 # We're not in PeerDirect mode
                 # (when we are the cleanup callback returns the item)
                 self._out_free_queue.put_nowait(out_item)
-            task = asyncio.create_task(chunk.send(streams, n_frames, self.engine.time_converter, self.engine.sensors))
-            task.add_done_callback(partial(self._chunk_finished, chunk))
+            task = asyncio.create_task(self._chunk_send_and_cleanup(streams, n_frames, chunk))
+            self.engine.add_service_task(task)
 
         if task:
             try:
```

Comment:
On line 829 there is a reference in a comment to chunk_finished, which needs to be updated.